src/dataframes.jl

```diff
@@ -21,12 +21,31 @@ function Plots.extractGroupArgs(group::Symbol, df::AbstractDataFrame, args...)
     Plots.extractGroupArgs(collect(df[group]))
 end
 
+# allows the passing of expressions including DataFrame columns as symbols
+function processExpr(expr, df)
+    arg = map(expr.args) do x
+        isa(x, Expr) && return processExpr(x, df)
+        if isa(x, QuoteNode) && isa(x.value, Symbol)
+            return collect(df[x.value])
```

Comment:
I have tried moving `eval` out as the last step but still doesn't work. I will keep looking at this
```julia
# allows the passing of expressions including DataFrame columns as symbols
function processExpr!(expr, df)
    arg = map(expr.args) do x
        isa(x, Expr) && processExpr!(x, df)
        if isa(x, QuoteNode) && isa(x.value, Symbol)
            return :(collect(df[:($(x.value))]))
        end
        x
    end
    expr.args = arg
end

# if a DataFrame is the first arg, lets swap symbols out for columns
@recipe function f(df::AbstractDataFrame, args...)
    # if any of these attributes are symbols, swap out for the df column
    for k in (:fillrange, :line_z, :marker_z, :markersize, :ribbon, :weights, :xerror, :yerror)
        if haskey(d, k)
            if isa(d[k], Expr)
                processExpr!(d[k], df)
                d[k] = eval(d[k])
            end
```